### DISCUSSION TITLE
extract source script from image (.bin / .img)

### DISCUSSION THREAD

**Discussion by picpic020960** [micropython/micropython#10077] [General]
Bonjour ,
I have corrupted my ESP32 , and no backup   (:-(().
I wish extract source script from the image (from esptool)
Exists tool for this ?
Thank's for help and Regards

**Comment by scruss** [comment]
There might be something useful in this thread: [Recovering files from a Pico file system - Raspberry Pi Forums](https://forums.raspberrypi.com/viewtopic.php?t=337154)

I know it refers to a Raspberry Pi Pico and you are using an ESP32, but the 'dumplfs' tool linked there looks like it might be able to dump littlefs filesystems embedded in a binary. Good luck! We've all¹ been there!

---
¹: if you haven't been there, it's that you haven't been there *yet* …

**Reply by picpic020960** [reply]
Thanks , I will test !

**Reply by jimmo** [reply]
Thanks @scruss -- yep this is exactly the process. Given a .bin file you can find the offset of the filesystem and then use the littlefs tools to access it. You might need to explicitly configure the block size.

I don't quite follow how you were able to get the .bin file if the ESP32 is already corrupted though.

**Reply by picpic020960** [reply]
With esptool , read_flash , no ?

**Reply by jimmo** [reply]
@picpic020960 Right, but if the device filesystem is corrupted, then so will the image that you read back with esptool.
